tests/ui/pattern/usefulness/empty-types.never_pats.stderr

```diff
@@ -430,7 +430,7 @@ note: `Option<!>` defined here
    = note: `!` is uninhabited but is not being matched by value, so a wildcard `_` is required
 help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
-LL ~         &None => {},
+LL ~         &None => {}
 LL +         &Some(!)
```

Comment:
Another case of already broken suggestion :dizzy_face:

Reply:
Given this, the second commit is incorrect. Fixing.

Reply:
Never types sure need some polish :-/

Except for better handling cases involving macros in patterns and or-patterns surrounded by parentheses (which is recovered syntax to begin with), the suggestions are now "correct".